service/controller/cluster.go

```diff
@@ -111,6 +113,9 @@ func NewCluster(config ClusterConfig) (*Cluster, error) {
 			NewRuntimeObjectFunc: func() runtime.Object {
 				return new(infrastructurev1alpha2.AWSCluster)
 			},
+			Selector: labels.SelectorFromSet(map[string]string{
+				label.OperatorVersion: project.Version(),
+			}),
```

Comment:
We can set a label selector to only reconcile objects for that operator version. Kubernetes ensures to not give us other CRs in that case.